### DISCUSSION TITLE
Using IOfflineStoreExtensions DefineTable

### DISCUSSION THREAD

**Discussion by henda79** [Azure/azure-mobile-apps#761] [Q&A]
Hi, I've recently started porting to MAUI... oh the joys.

I've took this opportunity to refactor some code. Previously our table names matched the model names, but now my model are suffixed with Dto, so `UserDto` for example and the table name is `Users`. So with this in mind, I define all my tables using:

`IOfflineStoreExtensions.DefineTable<UserDto>("Users");`

However I notice when calling the `GetOfflineTable` method, it prepares the SQL statement using the table name `userdto`. I've looked at the code I can see that `DatasyncContractresolver.ResolveTableName` only gets the table name from the attribute in the model, which is great, but the question here is what is the point in defining the tables with explicit names ?

What I'd like to see is the resolution of table names from here or the attributes. Correct me please if I've missed something :)

**Comment by adrianhall** [comment]
This feels like a bug - let's transition it to an issue.
